pages/verification/index.tsx

```diff
@@ -0,0 +1,169 @@
+                        generateAnonUser.mutate({ userId: localStorage.getItem(MessageEnum.targetUser) }, {
+                            onSuccess: (data) => {
+                                initChatAnon.mutate({
+                                    anon_user_info_color_code: data.anon_user_info_color_code,
+                                    anon_user_info_color_name: data.anon_user_info_color_name,
+                                    anon_user_info_emoji_code: data.anon_user_info_emoji_code,
+                                    anon_user_info_emoji_name: data.anon_user_info_emoji_name,
+                                    members: [member],
+                                    message: message,
+                                }, {
+                                    onSuccess: (data) => {
+                                        if (data) {
+                                            setIsLoading(false);
+                                            router.push('/message-sent');
+                                        }
+                                        localStorage.removeItem(MessageEnum.tempMessage);
+                                        localStorage.removeItem(MessageEnum.targetUser);
+                                    },
+                                    onError: (err) => {
+                                        setIsLoading(false);
+                                        console.error(err)
+                                        toast('We failed to send your message', {
+                                            autoClose: 3000,
+                                            type: 'error',
+                                        })
+                                    }
+                                })
+                            },
+                            onError: (err) => {
+                                setIsLoading(false);
+                                console.error(err)
+                                toast('We failed to send your message', {
+                                    autoClose: 3000,
+                                    type: 'error',
+                                })
+                            }
+                        })
+                    }
+                },
+                onError: (err) => {
+                    setIsLoading(false);
+                    console.error(err)
+                    toast('humanID verification failed, please try again', {
+                        autoClose: 3000,
+                        type: 'error',
+                    })
+                }
+            })
+        }
+        if (props.isFailedVerify) {
+            toast('humanID verification failed, please try again', {
+                autoClose: 3000,
+                type: 'error',
+            })
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a lot of logic inside this `useEffect` hook which makes it hard to read and maintain. Consider breaking it down into smaller functions or custom hooks. This would make the code easier to understand and test.



<!-- This is an auto-generated comment by OSS CodeRabbit -->